lib/widgets/profile.dart

```diff
@@ -0,0 +1,334 @@
+          final RealmDefaultExternalAccount? realmDefaultExternalAccount = store.realmDefaultExternalAccounts[fieldData["subtype"]];
+          if (realmDefaultExternalAccount != null) {
+            urlPattern = realmDefaultExternalAccount.urlPattern;
+          }
+        }
+        if (urlPattern != null) {
+          final url = urlPattern.replaceFirst("%(username)s", entry.text);
+          return Padding(
+            padding: const EdgeInsets.all(4),
+            child: ProfileDataTableLink(url: url, label: entry.text));
+        } else {
+          return Padding(
+            padding: const EdgeInsets.all(4),
+            child: Text(entry.text));
+        }
+      case CustomProfileFieldType.user:
+        // TODO(server): This is completely undocumented.  The key to
+        //   reverse-engineering it was:
+        //   https://github.com/zulip/zulip/blob/18230fcd9/static/js/settings_account.js#L247
+        final List<dynamic> userIds = jsonDecode(entry.text);
+        if (userIds.isEmpty) {
+          return const SizedBox.shrink();
+        }
+        return Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: userIds.whereType<num>()
+            .map((userId) => store.users[userId])
+            .whereType<User>()
+            .map((user) => ProfileDataTableUser(user: user)).toList());
+      case CustomProfileFieldType.date:
+      // TODO(server): The value's format is undocumented, but empirically
+      //   it's a date in ISO format, like 2000-01-01.
+      // That's readable as is, but:
+      // TODO format this date using user's locale.
+        return Padding(
+          padding: const EdgeInsets.all(4),
+          child: Text(entry.text));
+      case CustomProfileFieldType.shortText:
+      case CustomProfileFieldType.longText:
+      case CustomProfileFieldType.pronouns:
+      case CustomProfileFieldType.unknown:
+        return Padding(
+          padding: const EdgeInsets.all(4),
+          child: Text(entry.text));
+    }
+  }
+}
+
+class CustomProfileFieldDisplayEntry {
+  final String label;
+  final String text;
+  final CustomProfileFieldType type;
+  final String fieldData;
+
+  CustomProfileFieldDisplayEntry({required this.fieldData, required this.label, required this.text, required this.type});
+}
+
+class ProfileDataTableLink extends StatelessWidget {
+  const ProfileDataTableLink({super.key, required this.label, required this.url});
+
+  final String label;
+  final String url;
+
+  @override
+  Widget build(BuildContext context) {
+    final Uri? parsedUrl = Uri.tryParse(url);
+    if (parsedUrl == null) {
+      return Text(label);
+    }
+    return MouseRegion(
+      cursor: SystemMouseCursors.click,
+      child: GestureDetector(
+        onTap: () async {
+          await ZulipBinding.instance.launchUrl(parsedUrl);
+        },
+        child: Text(label,
+          style: TextStyle(color: const HSLColor.fromAHSL(1, 200, 1, 0.4).toColor()))));
```

Comment:
It'd be good to dedupe this color between here and the message content, too.

Hmm, in fact — maybe the best way to handle these links is just to treat them *as Zulip content*. So just construct an appropriate content node and pass that to a widget from `widgets/content.dart`. Something like `Paragraph(node: ParagraphNode(nodes: [LinkNode(url: entry.text, label: entry.text)]))`… ok, to actually form it properly including `ParagraphNode.links`, it'd be like:
```dart
  final linkNode = LinkNode(url: entry.text, label: entry.text);
  return Paragraph(node: ParagraphNode(nodes: [linkNode], links: [linkNode]));
```